### PR TITLE
Unprivileged install for openSUSE/SUSE

### PR DESCRIPTION
**Pull request by mslacken**
Adds openSUSE/SUSE distros to `tools/install-unprivileged.sh`.
Installation can be done via KOJI rpms or the rpms from the openBuild service.

### CODE REVIEW

**Comment by DrDaveD** [comment]
Thanks!  In general this looks good.  Please fix the shellcheck ci test.  If you think the message is invalid you can always disable it with a `# shellcheck disable=` comment.  Also please re-do the commits with `git -s` to make the DCO happy.  The rocky9 failure is a known temporary failure that should be fixed soon.

Add to the usage message the fact that current suse distributions are also supported.  For the "-o" option please make it clear that it only applies to opensuse distributions.

**Comment by mslacken** [comment]
Added the requested lines of explanation.

File: tools/install-unprivileged.sh [hunk 1]
```diff
@@ -156,7 +196,7 @@ LASTPKGS=""
 latesturl()
 {
 	typeset URL="$1"
-	if [ "$3" = true ]; then
+	if [ "$3" == true ]; then
```
Comment (DrDaveD) [inline tools/install-unprivileged.sh:199]: If you make this kind of change please make it consistently.  I see for example below `[ "$4" = true ]`.

File: tools/install-unprivileged.sh [hunk 2]
```diff
@@ -179,11 +219,11 @@ latesturl()
 	fi
 }
 
-APPTAINERURL="https://kojipkgs.fedoraproject.org/packages/apptainer/"
 LOCALAPPTAINER=false
-if [ -z "$VERSION" ]; then
+#if test -z "$VERSION" || ! test -z $OBSURL ; then
+if test -z "$VERSION" || [[ "$KOJI" == "false" ]]; then
```
Comment (DrDaveD) [inline tools/install-unprivileged.sh:224]: Let's not leave this comment, and don't make the style inconsistent by changing `[ -z` to `test`.

File: tools/install-unprivileged.sh [hunk 3]
```diff
@@ -284,7 +337,12 @@ cd ..
 
 set -e
 
+# remove .build-id files seen on el8 or later
+rm -rf lib/.build-id
+rmdir lib 2>/dev/null || true
```
Comment (DrDaveD) [inline tools/install-unprivileged.sh:342]: Since this is my change, I think you should instead rebase on the current main and force-push.

**Review by DrDaveD — CHANGES_REQUESTED** [review]
Please address the new failure on ubuntu in the rocky9 CI test.

And that reminds me: please add additional invocations of TEST_TYPE=unpriv in the centos7, rocky8, and rocky9 tests for the different SUSE distributions that have dockerhub containers available.

**Comment by mslacken** [comment]
Checks pass, but I had to remove tumbleweed as rolling distributions show sometimes strange errors.
I will check this more closely next year

**Review by DrDaveD — CHANGES_REQUESTED** [review]
If I test this with the "-o" option (on an opensuse/leap docker container) I get
```
Extracting https://download.opensuse.org/repositories/network:/cluster/openSUSE_Leap_15.4/x86_64/apptainer-1.1.2-lp154.30.17.x86_64.rpm
argument is not an RPM package
cpio: premature end of archive
Error extracting https://download.opensuse.org/repositories/network:/cluster/openSUSE_Leap_15.4/x86_64/apptainer-1.1.2-lp154.30.17.x86_64.rpm
```
I expect that since it isn't 1.1.4 that the resultant install wouldn't work, but that URL gets a 404 Not Found error and that's not right.

**Comment by DrDaveD** [comment]
@mslacken Reminder, this is waiting on you.

**Comment by mslacken** [comment]
Sorry for the late reply, but I was on holidays.
Basically always only *one* apptainer version is available with the open build service as older versions get deleted as soon as a new version is accepted. 
I have updated the package in the devel repo (https://build.opensuse.org/repositories/network:/cluster)  so that only 1.1.4 is now available.
I think this solves the problem.

**Comment by DrDaveD** [comment]
That still doesn't quite do it.  I still get the same message.  I see that https://download.opensuse.org/repositories/network:/cluster/openSUSE_Leap_15.4/x86_64, the one being used, still has apptainer-1.1.2, but that apptainer-1.1.4 is in https://mirrorcache-us.opensuse.org/repositories/network:/cluster/15.4/x86_64/.  Does the URL need to change?  Oh I see it has already been changed in $OSREPOURL but it hasn't been changed in $EPELREPOURL.  Yes, if I make that change, it works.

I want to also make a few style changes, so I'll go ahead and push a fix to the PR branch.

**Comment by DrDaveD** [comment]
Oddly, if you look at https://download.opensuse.org/repositories/network:/cluster there's no "15.4" shown, but yet if you just add it to the URL it gets you to the more up to date repository.  Something is strange there.

**Comment by DrDaveD** [comment]
> Oddly, if you look at https://download.opensuse.org/repositories/network:/cluster there's no "15.4" shown, but yet if you just add it to the URL it gets you to the more up to date repository. Something is strange there.

Actually, it seems that sometimes it redirects me to https://mirrorcache-us.opensuse.org/repositories/network:/cluster and it is only then that I don't see the "15.4".  I don't understand why sometimes I get redirected and sometimes I don't.

**Review by DrDaveD — APPROVED** [review]

**Comment by DrDaveD** [comment]
@mslacken This looks good to me now.  I will wait to merge until you tell me what you think about the additional changes I made.

**Comment by mslacken** [comment]
Thanks for your additional work.

**Pull request merged**